src/Compilers/Test/Utilities/CSharp/CompilationTestUtils.cs

```diff
@@ -306,31 +306,19 @@ internal static void VerifyTypes(this CSharpCompilation compilation, SyntaxTree
             var annotationsByMethod = allAnnotations.GroupBy(annotation => annotation.Expression.Ancestors().OfType<BaseMethodDeclarationSyntax>().First()).ToArray();
             foreach (var annotations in annotationsByMethod)
             {
-                var method = (MethodSymbol)model.GetDeclaredSymbol(annotations.Key);
-
-                var diagnostics = DiagnosticBag.GetInstance();
-                var block = MethodCompiler.BindMethodBody(method, new TypeCompilationState(method.ContainingType, compilation, null), diagnostics);
-                var rewritten = NullableWalker.AnalyzeAndRewrite(
-                    compilation,
-                    method,
-                    block,
-                    diagnostics);
-                var dictionary = NullabilityRetriever.BuildMap(rewritten);
-                diagnostics.Free();
+                var methodSyntax = annotations.Key;
+                var method = model.GetDeclaredSymbol(methodSyntax);
+
                 var expectedTypes = annotations.SelectAsArray(annotation => annotation.Text);
-                var actualTypes = annotations.SelectAsArray(annotation => toDisplayString(annotation.Expression));
+                var actualTypes = annotations.SelectAsArray(annotation =>
+                {
+                    var typeInfo = model.GetTypeInfo(annotation.Expression);
+                    Assert.NotEqual(CodeAnalysis.NullableAnnotation.NotApplicable, typeInfo.Nullability.Annotation);
+                    Assert.NotEqual(CodeAnalysis.NullableFlowState.NotApplicable, typeInfo.Nullability.FlowState);
+                    return new TypeWithState((TypeSymbol)typeInfo.Type, (NullableFlowState)(typeInfo.Nullability.FlowState - 1)).ToTypeWithAnnotations().ToDisplayString(TypeWithAnnotations.TestDisplayFormat);
```

Comment:
>new TypeWithState((TypeSymbol)typeInfo.Type, (NullableFlowState)(typeInfo.Nullability.FlowState - 1)).ToTypeWithAnnotations() [](start = 27, length = 125)

Is this conversion needed or could we call the following?
```
typeInfo.Type.ToDisplayString(
    typeInfo.Nullability.FlowState,
    TypeWithAnnotations.TestDisplayFormat);
```

Reply:
Unfortunately, part of the test formatting code currently lives in `TypeWithAnnotations.ToDisplayString`. Correcting this is part of my planned refactor of the symbol display apis.

---
In reply to: [269223710](https://github.com/dotnet/roslyn/pull/34429#discussion_r269223710) [](ancestors = 269223710)